OMCompiler/Compiler/SimCode/SimCodeMain.mo

```diff
@@ -1328,7 +1328,9 @@ algorithm
     modelInfo := SimCodeUtil.addNumEqns(modelInfo, uniqueEqIndex-listLength(jacobianEquations));
 
     // update hash table
-    crefToSimVarHT := SimCodeUtil.createCrefToSimVarHT(modelInfo);
```

Comment:
This here is the problem. If you un-comment this line then this will discard the things added to the crefToSimVarHT hashtable. 

When we comment it out then we get some variables generated with wrong indices. You can try simulating  
[testDAEp9.mos](testsuite/simulation/modelica/daemode/testDAEp9.mos). You will see that in the first equation in the generated `testDAE.p9_16dae.c` file the variables w[2] (variable) and x[2] (STATE(1)) are both generated pointing to the same entry `data->localData[0]->realVars[1]`. 

Clearly w[2] is generated wrong. If you uncomment the above mentioned cration on new hastable then these variables get generated correctly to `data->localData[0]->realVars[11] /* w[2] variable */ `and `data->localData[0]->realVars[1] /* x[2] STATE(1) */`